diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7ef8f03f..4096b037 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required (VERSION 3.0)
+cmake_minimum_required (VERSION 3.1.2)
 
 project (wali)
 set(CMAKE_PROJECT_NAME "wali")
@@ -19,12 +19,15 @@ if(NOT DEFINED CMAKE_CXX_EXTENSIONS)
 set(CMAKE_CXX_EXTENSIONS OFF)
 endif(NOT DEFINED CMAKE_CXX_EXTENSIONS)
 
-if(NOT DEFINED LLVM_MAIN_SRC_DIR)
-  # Find an include llvm if we build out of tree
-  find_package(LLVM REQUIRED CONFIG)
-  find_library(LLVM_LIBRARY NAMES llvm)
-  include_directories(${LLVM_INCLUDE_DIRS})
-  link_directories(${LLVM_LIB_PATH} ${LLVM_LIBRARY_DIRS})
+# setup conan for dependecy resolution
+if (EXISTS ${CMAKE_BINARY_DIR}/conan_paths.cmake)
+  set(CONAN ON)
+  include(${CMAKE_BINARY_DIR}/conan_paths.cmake)
+  find_package(llvm-core 12 REQUIRED)
+else()
+  set(CONAN OFF)
+  find_package(LLVM 12 REQUIRED)
 endif()
+message("[CMakeLists] Conan usage: " ${CONAN})
 
 add_subdirectory(Source)
diff --git a/Source/wali/lib/CMakeLists.txt b/Source/wali/lib/CMakeLists.txt
index 0b073f74..c1f17c15 100644
--- a/Source/wali/lib/CMakeLists.txt
+++ b/Source/wali/lib/CMakeLists.txt
@@ -19,22 +19,21 @@ add_library(wali # Name der Bibliothek
 		${WALI_SRC} # Quellcode der zur dieser Bibliothek gehören soll
 )
 
-set(LLVM_LINK_COMPONENTS
-  Support
-  Core
-)
-if(USE_LLVM_FAT_LIB)
-  llvm_config(wali USE_SHARED ${LLVM_LINK_COMPONENTS})
-else()
-  llvm_config(wali ${LLVM_LINK_COMPONENTS})
-endif()
-
 # Falls nötig und die Bibliothek selbst Abhangigkeiten besitzt,
 # kann man dieser hiermit dazu linken. Man sollte allerdings
 # soweit es geht darauf verzichten, da man ansonsten schnell in
 # der 'dependency hell' landet.
-# target_link_libraries(test_all
-# )
+if (CONAN)
+	target_include_directories(wali PRIVATE ${llvm-core_INCLUDE_DIRS})
+	target_link_libraries(wali PRIVATE llvm-core::llvm-core)
+else()
+	target_include_directories(wali PRIVATE ${LLVM_INCLUDE_DIRS})
+	if(USE_LLVM_FAT_LIB)
+		llvm_config(wali USE_SHARED Core Support)
+	else()
+		llvm_config(wali Core Support)
+	endif()
+endif()
 
 set_target_properties(wali
 	PROPERTIES